lib/pbench/test/unit/server/test_requests.py

```diff
@@ -59,19 +52,29 @@ def test_empty_url_path(client, caplog):
         assert caplog.records[0].levelname == "WARNING"
 
     @staticmethod
-    def test_bad_request(client, caplog, mocked_responses):
-        mocked_responses.add(
-            responses.POST,
-            "http://elasticsearch.example.com:7080/some_invalid_url",
-            status=400,
+    def test_bad_request(client, caplog, requests_mock):
+        requests_mock.post(
+            "http://elasticsearch.example.com:7080/some_invalid_url", status_code=400
         )
+        # mocked_responses.add(
+        #     responses.POST,
+        #     "http://elasticsearch.example.com:7080/some_invalid_url",
+        #     status=400,
+        # )
         response = client.post(
             f"{client.config['REST_URI']}/elasticsearch",
             json={"indices": "some_invalid_url", "payload": '{ "babble": "42" }'},
         )
         assert response.status_code == 400
         assert response.json is None
-        assert len(caplog.records) == 0
+
+        # This is a bit awkward, but the requests_mock throws in its own
+        # DEBUG log record to announce the POST; so allow it. (But don't
+        # fail if it's missing.)
+        if len(caplog.records) > 0:
+            assert len(caplog.records) == 1
+            assert caplog.records[0].levelname == "DEBUG"
+            assert caplog.records[0].name == "requests_mock.adapter"
```

Comment:
And the mock requests for some reason is generating its own `DEBUG` log here; so I just accounted for it. (Sigh.)